zerver/lib/user_groups.py

```diff
@@ -185,35 +186,51 @@ def access_user_group_for_deactivation(
             **{f"{setting_name}__in": setting_group_ids_using_deactivating_user_group}
         )
 
-    if (
-        Stream.objects.filter(realm_id=user_group.realm_id, deactivated=False)
-        .filter(stream_setting_query)
-        .exists()
+    for stream in Stream.objects.filter(realm_id=user_group.realm_id, deactivated=False).filter(
+        stream_setting_query
```

Comment:
`setting_group_ids_using_deactivating_user_group` likely needs to be converted to a `Set` for performance reasons.

Reply:
Updated.